dao/src/main/scala/za/co/absa/enceladus/dao/auth/MenasCredentials.scala

```diff
@@ -42,7 +54,25 @@ object MenasPlainCredentials {
   def fromFile(path: String)(implicit spark: SparkSession): MenasPlainCredentials = {
     val fs =  FileSystemUtils.getFileSystemFromPath(path)(spark.sparkContext.hadoopConfiguration)
 
-    val conf = ConfigFactory.parseString(HadoopFsUtils.getOrCreate(fs).getLocalOrDistributedFileContent(path))
+    val conf = fs match {
+      case _: DistributedFileSystem | LocalFileSystem => {
+        val str = HadoopFsUtils.getOrCreate(fs).getLocalOrDistributedFileContent(path)
+        log.info(str)
+        ConfigFactory.parseString(str)
+      }
+      case _ => {
+        val s3Path: SimpleS3Location = path.toSimpleS3Location.get
+
+        val client = new AmazonS3Client()
+        val s3Object = client.getObject(s3Path.bucketName, s3Path.path)
+        val content: S3ObjectInputStream = s3Object.getObjectContent
+        val str1 = scala.io.Source.fromInputStream(content).mkString
```

Comment:
`str1` sounds wrong. How about:

```suggestion
        val inputStream: S3ObjectInputStream = s3Object.getObjectContent
        val content = scala.io.Source.fromInputStream(inputStream).mkString
```

or 

```suggestion
        val is: S3ObjectInputStream = s3Object.getObjectContent
        val content = scala.io.Source.fromInputStream(is).mkString
```

Reply:
True, WIP currently